beacon_node/beacon_chain/src/data_availability_checker.rs

```diff
@@ -192,20 +212,18 @@ impl<T: BeaconChainTypes> DataAvailabilityChecker<T> {
         )
     }
 
-    pub fn put_gossip_data_columns(
+    pub fn put_gossip_data_column(
         &self,
-        slot: Slot,
-        block_root: Hash256,
-        gossip_data_columns: Vec<GossipVerifiedDataColumn<T>>,
+        gossip_data_column: GossipVerifiedDataColumn<T>,
```

Comment:
I changed this in #5806 to avoid reconstruction for proposer nodes. Is this scenario handled?

Reply:
Right, thanks for pointing that out I wasn't aware. `publish_block` also ensures that all columns are from the same block. Do you think we need the `*SameBlock` type here too? Switching to a single item was just a way to get around having to create a `GossipVerifiedDataColumnsSameBlock` or similar.

Reply:
I think it was clear that these are from the same block given the `block_root` and `slot` parameters. Is it necessary to introduce a new type? as we always either pass a single data column around, or pass a list of data columns for the same block. 

Could we maybe delay these kind of refactor in separate future PRs. Feels like it makes conga line PRs difficult to manage (we'll also have to re-work all subsequent PRs)